### PR TITLE
Assets generation

### LINKED ISSUE #290: Problem with Contributing Guideline set up
- Versions:
  - Go: `go version go1.11.1 darwin/amd64`
  - Yarn: `1.10.1`
- I was going through the [development part of Contributing Guidelines](https://github.com/src-d/landing/blob/master/CONTRIBUTING.md#development) to run the server locally so I could make a make some changes to the project for a PR, and kept running into an error.
- All of the previous requirements listed in the guidelines are installed but running the `make serve` command shows an error:

```bash
alicia~» go version                                                 [18:39:01]
go version go1.11.1 darwin/amd64
alicia~» go env GOPATH;                                             [18:39:16]
/Users/aliciakozlowski/go
alicia~» go get github.com/src-d/landing/...                        [18:39:23]
alicia~» cd $GOPATH/src/github.com/src-d/landing                    [18:39:43]
aliciagithub.com/src-d/landing(master)» yarn --version;             [18:40:05]
1.10.1
aliciagithub.com/src-d/landing(master)» make build                  [18:41:03]
git clone --quiet --depth 1 https://github.com/src-d/ci.git /Users/aliciakozlowski/go/src/github.com/src-d/landing/.ci;
Downloading https://github.com/spf13/hugo/releases/download/v0.21/hugo_0.21_macOS-64bit.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   151    0   151    0     0    308      0 --:--:-- --:--:-- --:--:--   308
100   618    0   618    0     0    900      0 --:--:-- --:--:-- --:--:--  603k
100 4435k  100 4435k    0     0  1901k      0  0:00:02  0:00:02 --:--:-- 5716k
/Users/aliciakozlowski/go/src/github.com/src-d/landing/.hugo/hugo --config=hugo.config.yaml --destination=public --baseURL=//sourced.tech
Started building sites ...
Built site for language en:
0 draft content
0 future content
0 expired content
8 regular pages created
8 other pages created
0 non-page files copied
0 paginator pages created
1 classes created
0 categories created
total in 14 ms
aliciagithub.com/src-d/landing(master)» LANDING_URL=//localhost PORT=8081 make serve
rm -rf /Users/aliciakozlowski/go/src/github.com/src-d/landing/.hugo
Downloading https://github.com/spf13/hugo/releases/download/v0.21/hugo_0.21_macOS-64bit.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   151    0   151    0     0    296      0 --:--:-- --:--:-- --:--:--   296
100   618    0   618    0     0    899      0 --:--:-- --:--:-- --:--:--   899
100 4435k  100 4435k    0     0  1996k      0  0:00:02  0:00:02 --:--:-- 4281k
yarn run serve
yarn run v1.10.1
$ concurrently --kill-others "yarn run webpack-watcher" "make hugo-server" "yarn run api-run"
/bin/sh: concurrently: command not found
error Command failed with exit code 127.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
make: *** [serve] Error 1
aliciagithub.com/src-d/landing(master)» yarn start                                         [18:50:15]
yarn run v1.10.1
$ node_modules/.bin/termax server
/bin/sh: node_modules/.bin/termax: No such file or directory
error Command failed with exit code 127.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
aliciagithub.com/src-d/landing(master)» make serve                                         [18:51:12]
rm -rf /Users/aliciakozlowski/go/src/github.com/src-d/landing/.hugo
Downloading https://github.com/spf13/hugo/releases/download/v0.21/hugo_0.21_macOS-64bit.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   151    0   151    0     0    250      0 --:--:-- --:--:-- --:--:--   250
100   618    0   618    0     0    789      0 --:--:-- --:--:-- --:--:--  603k
100 4435k  100 4435k    0     0  1700k      0  0:00:02  0:00:02 --:--:-- 5310k
yarn run serve
yarn run v1.10.1
$ concurrently --kill-others "yarn run webpack-watcher" "make hugo-server" "yarn run api-run"
/bin/sh: concurrently: command not found
error Command failed with exit code 127.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
make: *** [serve] Error 1
aliciagithub.com/src-d/landing(master)» yarn install concurrently                          [18:51:25]
yarn install v1.10.1
error `install` has been replaced with `add` to add new dependencies. Run "yarn add concurrently" instead.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
aliciagithub.com/src-d/landing(master)» yarn add concurrently                              [18:54:27]
yarn add v1.10.1
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
[4/4] 📃  Building fresh packages...
[1/2] ⡀ fsevents
warning Error running install script for optional dependency: "/Users/aliciakozlowski/go/src/github.com/src-d/landing/node_modules/fsevents: Command failed.
Exit code: 1
Command: node install
Arguments: 
Directory: /Users/aliciakozlowski/go/src/github.com/src-d/landing/node_modules/fsevents
Output:

---------- a bunch of things that probably don´t matter -----------

4 warnings and 1 error generated.
make: *** [Release/obj.target/binding/src/binding.o] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/Users/aliciakozlowski/go/src/github.com/src-d/landing/node_modules/node-gyp/lib/build.js:258:23)
gyp ERR! stack     at ChildProcess.emit (events.js:182:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:240:12)
gyp ERR! System Darwin 17.7.0
gyp ERR! command "/usr/local/Cellar/node/10.11.0/bin/node" "/Users/aliciakozlowski/go/src/github.com/src-d/landing/node_modules/node-gyp/bin/node-gyp.js" "rebuild" "--verbose" "--libsass_ext=" "--libsass_cflags=" "--libsass_ldflags=" "--libsass_library="
gyp ERR! cwd /Users/aliciakozlowski/go/src/github.com/src-d/landing/node_modules/node-sass
gyp ERR! node -v v10.11.0
gyp ERR! node-gyp -v v3.6.2
gyp ERR! not ok 
Build failed with error code: 1

aliciagithub.com/src-d/landing(master)» make serve                                         [19:01:39]
rm -rf /Users/aliciakozlowski/go/src/github.com/src-d/landing/.hugo
Downloading https://github.com/spf13/hugo/releases/download/v0.21/hugo_0.21_macOS-64bit.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   151    0   151    0     0    287      0 --:--:-- --:--:-- --:--:--   287
100   618    0   618    0     0    873      0 --:--:-- --:--:-- --:--:--   873
100 4435k  100 4435k    0     0  1119k      0  0:00:03  0:00:03 --:--:-- 1622k
yarn run serve
yarn run v1.10.1
$ concurrently --kill-others "yarn run webpack-watcher" "make hugo-server" "yarn run api-run"
$ go run api/api.go
$ webpack --config=webpack.config.js --watch --debug --verbose --progress --colors
[0]  10% building modules 1/1 modules 0 active                                         
[0] Webpack is watching the files...
[0] 
 11% building modules 16/19 modules 3 active ...dules/lightbox2/dist/css/lightbox.css/Users/aliciakozlowski/go/src/github.com/src-d/landing/.hugo/hugo server --config=hugo.config.yaml --destination=public --port=80 --baseURL=//sourced.tech --watch --buildDrafts
[1] Error: Server startup failed: listen tcp 127.0.0.1:80: bind: permission denied
[1] make[1]: *** [hugo-server] Error 255
[1] make hugo-server exited with code 2
--> Sending SIGTERM to other processes..
[2] [GIN-debug] [WARNING] Creating an Engine instance with the Logger and Recovery middleware already attached.
[2] 
[2] [GIN-debug] [WARNING] Running in "debug" mode. Switch to "release" mode in production.
[2]  - using env:	export GIN_MODE=release
[2]  - using code:	gin.SetMode(gin.ReleaseMode)
[2] 
[2] [GIN-debug] GET    /                         --> github.com/src-d/landing/api/handlers.Head.func1 (4 handlers)
[2] [GIN-debug] GET    /api/posts/:kind          --> github.com/erizocosmico/gin-cache.CachePage.func1 (4 handlers)
[2] [GIN-debug] GET    /api/positions            --> github.com/erizocosmico/gin-cache.CachePage.func1 (4 handlers)
[2] [GIN-debug] POST   /api/invite               --> github.com/src-d/landing/api/handlers.SlackInvite.func1 (4 handlers)
[2] [GIN-debug] Listening and serving HTTP on :8080
[0] yarn run webpack-watcher exited with code 1
--> Sending SIGTERM to other processes..
[2] yarn run api-run exited with code 1
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
make: *** [serve] Error 1
```

### LINKED ISSUE #291: Missing yarn dependencies in Makefile
Commit da3f0fd removed the `js-dependencies` execution from the Makefile, but it is needed, at least, by `make serve`.

Related issue: #290.

### LINKED ISSUE #293: Automate sass generation
We have sass and css commited in the repo. But the css is not generated automatically, and we don't document how to do it in [contributing](https://github.com/src-d/landing/blob/master/CONTRIBUTING.md). Maybe something like

- Install sass: https://sass-lang.com/install
- generate css from scss:
```bash
cd static/scss
sass style.scss style.css
```

I think we should have a check in CI for out of sync css, or remove it from the repo and generate it on build.

### LINKED ISSUE #311: Static generation was removed after the redesign
causing #290 #293 #291

The generation of the static files should be restored

### PR DESCRIPTION
**Pull request by dpordomingo**
fix #311 
fix #290 
fix #291
fix  #293 
supersedes #301 #313 
blocks #317 #318 

Already available to be validated at Staging http://landing-staging.srcd.run

Changes:
- 42b674d Autogenerate `css` and `js` from sources (#311 #293 )
- 998c0f3 Update `CONTRIBUTING.md` to show how to run `yarn start` (#290 #291)
- 015a2ee Rearrange `static/assets/*` to avoid duplicities
- 42b674d Use `jquery`, `bootstrap` and `slick` from `npm`
- 417e10b Clean and update old dependencies

### CODE REVIEW

**Comment by dpordomingo** [comment]
@eiso I added you to the reviewers list because you performed the integration of the design and took care of the deployment of everything (if I'm not wrong).
I worked with the assumption that the "autogeneration" of the assets were dropped to reduce the complexity during the redesign, but it should be restored at some point; feel free to purpose any change.

File: src/js/index.js [hunk 1]
```diff
@@ -1,195 +1,79 @@
...
-  const menus = Array.from(document.querySelectorAll('#menu > ul'));
-  navs.forEach((n) => {
-    const to = n.getAttribute('data-nav-to');
-    n.addEventListener('click', (e) => {
-      e.preventDefault();
-
-      menus.forEach(m => m.classList.remove('visible'));
-      const menuTo = document.getElementById(`menu-${to}`) || mainMenu;
-      menuTo.classList.add('visible');
+// $(function () {
```
Comment (smacker) [inline src/js/index.js]: commented code 🤔

**Comment by carlosms** [comment]
What about all the functions removed from `index.js`, like [renderHorizontalSlackForms](https://github.com/src-d/landing/pull/316/files#diff-99a85c3932c0670278158f38d818e0efL86)? There were not used at all, any of them? Were those part of the old web functionality and it's not needed after the redesign, or what's the reasoning here?

**Comment by dpordomingo** [comment]
I didn't want to delete old code in this iteration.
There are many other leftovers, like `slackin`, `docsrv` interface, old templates and old data...
I'd go over them in a next iteration if you agree @carlosms

**Comment by carlosms** [comment]
Actually I was asking the opposite. I see a lot of code removed from `index.js`, and I assume it's safe because it's from leftovers, but I wanted to confirm.

**Comment by dpordomingo** [comment]
I misunderstood you.

The whole content of `src/js` was unused after the redesign, and it could be deleted. What I did is to copy the content of `static/js-load/script.js` into `src/js/index.js` replacing the old and unused code.
In the next iteration I'll clean the rest of the `src/js` directory and all the other unused code everywhere.

Is it ok to you @carlosms

**Comment by carlosms** [comment]
That sounds like a good plan, thank you for confirming it

**Pull request merged**